### PR TITLE
Handling undefined/null data

### PR DESCRIPTION
**Pull request by mgibas**
Without this check Validator throws exception:
```
Cannot convert undefined or null to object at hasOwnProperty
```

### CODE REVIEW

File: spec/validator.js [hunk 1]
```diff
@@ -50,4 +50,14 @@ describe('Validator constructor', function() {
   it('should have a check method', function() {
     expect(validator.check).to.be.a.function;
   });
+
+  it('should handle undefined data', function(){
+    validator = new Validator(undefined, { name: 'required' })
+    validator.check();
```
Comment (iamdtang) [inline spec/validator.js]: I think there should be an assertion here that verifies the validation has failed. What do you think @garygreen?

File: spec/validator.js [hunk 2]
```diff
@@ -50,4 +50,14 @@ describe('Validator constructor', function() {
...
+  })
+
+  it('should handle null data', function(){
+    validator = new Validator(null, { name: 'required' })
+    validator.check();
```
Comment (iamdtang) [inline spec/validator.js]: Same as above.

File: spec/validator.js [hunk 3]
```diff
@@ -50,4 +50,14 @@ describe('Validator constructor', function() {
   it('should have a check method', function() {
     expect(validator.check).to.be.a.function;
   });
+
+  it('should handle undefined data', function(){
+    validator = new Validator(undefined, { name: 'required' })
```
Comment (iamdtang) [inline spec/validator.js]: missing var

File: spec/validator.js [hunk 4]
```diff
@@ -50,4 +50,14 @@ describe('Validator constructor', function() {
...
+    validator.check();
+  })
+
+  it('should handle null data', function(){
+    validator = new Validator(null, { name: 'required' })
```
Comment (iamdtang) [inline spec/validator.js]: missing var

File: spec/validator.js [hunk 5]
```diff
@@ -50,4 +50,14 @@ describe('Validator constructor', function() {
   it('should have a check method', function() {
     expect(validator.check).to.be.a.function;
   });
+
+  it('should handle undefined data', function(){
+    validator = new Validator(undefined, { name: 'required' })
+    validator.check();
+  })
+
+  it('should handle null data', function(){
```
Comment (iamdtang) [inline spec/validator.js]: add a space before the  {

**Review by iamdtang — CHANGES_REQUESTED** [review]

File: spec/validator.js (hunk 2 above)
Comment (mgibas) [reply spec/validator.js]: fixed

File: spec/validator.js (hunk 4 above)
Comment (mgibas) [reply spec/validator.js]: fixed

File: spec/validator.js (hunk 5 above)
Comment (mgibas) [reply spec/validator.js]: fixed

File: spec/validator.js (hunk 3 above)
Comment (mgibas) [reply spec/validator.js]: fixed

File: spec/validator.js [hunk 6]
```diff
@@ -50,4 +50,14 @@ describe('Validator constructor', function() {
   it('should have a check method', function() {
     expect(validator.check).to.be.a.function;
   });
+
+  it('should handle undefined data', function() {
+    var validator = new Validator(undefined, { name: 'required' })
+    validator.check();
+  })
```
Comment (iamdtang) [inline spec/validator.js]: add a semicolon please

File: spec/validator.js [hunk 7]
```diff
@@ -50,4 +50,14 @@ describe('Validator constructor', function() {
...
+
+  it('should handle null data', function() {
+    var validator = new Validator(null, { name: 'required' })
+    validator.check();
+  })
```
Comment (iamdtang) [inline spec/validator.js]: add a semicolon please

File: spec/validator.js [hunk 8]
```diff
@@ -50,4 +50,14 @@ describe('Validator constructor', function() {
   it('should have a check method', function() {
     expect(validator.check).to.be.a.function;
   });
+
+  it('should handle undefined data', function() {
+    var validator = new Validator(undefined, { name: 'required' })
+    validator.check();
```
Comment (iamdtang) [inline spec/validator.js]: I don't think check is part of the public API. How about using something like this: http://chaijs.com/api/bdd/#method_throw to verify that the validator doesn't throw when fails() is called?

File: spec/validator.js [hunk 9]
```diff
@@ -50,4 +50,14 @@ describe('Validator constructor', function() {
...
+  })
+
+  it('should handle null data', function() {
+    var validator = new Validator(null, { name: 'required' })
+    validator.check();
```
Comment (iamdtang) [inline spec/validator.js]: same as previous comment about check()

**Review by iamdtang — CHANGES_REQUESTED** [review]

File: spec/validator.js (hunk 6 above)
Comment (mgibas) [reply spec/validator.js]: That made me think: why the heck my linter stoped working?! :)

File: spec/validator.js (hunk 8 above)
Comment (mgibas) [reply spec/validator.js]: Yeah - thats much better

Comment (mgibas) [reply spec/validator.js]: There is some weird thing with this:
`expect(fn).to.throw(Error)`
its calling asserted method with wrong scope which screws up `this` inside validator instance and gives me this error:
`AssertionError: expected [Function: fails] to not throw 'Error' but 'TypeError: this._checkAsync is not a function' was thrown`.
I switched to a `fails()` method though.

**Pull request merged**